### DISCUSSION TITLE
Add query parameters to select statements

### DISCUSSION THREAD

**Discussion by FarhanSajid1** [aarondl/sqlboiler#1002] [General]
Hey there, I've been using sqlboiler for a while and it's really great. I had a question regarding parameter usage for `qm.SELECT`  https://github.com/volatiletech/sqlboiler/blob/v3.7.1/queries/qm/query_mods.go#L224, I see that it does not take in query parameters like the other query modifiers, and that makes sense for almost all use cases except for doing subqueries with parameters.

For example:
```
qm.Select("id", (SELECT count(*) from second_table WHERE temp_id = ?) as "counter"`, user_id)),
```

If we want to do this now, we have to use the `qm.Raw` + bind.

If I'm missing something, please let me know.

**Comment by aarondl** [comment]
Not missing anything. Subqueries start to get out of the realm of "boilerplate" sql queries and into things that are more complicated. It never set out to do subqueries and so currently support is weak.
